src/ui/filechooserwidget.cpp

```diff
@@ -0,0 +1,107 @@
+#include "filechooserwidget.h"
+
+#include <QFileDialog>
+#include <QFileInfo>
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QLineEdit>
+#include <QPushButton>
+FileChooserWidget::FileChooserWidget(QWidget* parent)
+    : QWidget(parent),
+      layout_(new QHBoxLayout(this)),
+      pathEdit_(new QLineEdit(this)),
+      mode_(Mode::Directory),
+      filter_(""),
+      openDirPath_("") {
+  _init("");
```

Comment:
`_init()` does real work here so should not be called from the constructor. If you really want to call it from the constructor you can probably use constructor chaining instead.

Reply:
`constructor chaining`, first time I hear that... I gonna have a look.
It's not real work, it's just there to factorize code of the 3 different constructors.
It is not really needed for this project but I don't see the benefit to narrow it down.

Reply:
following the remark from [this thread](https://stackoverflow.com/a/33275207) 

> However, there is a severe limitation that a constructor that calls another constructor is not allowed to initialize any other members. So you cannot do the following with a delegating constructor:

I find that constructor chaining is not a good idea. It is slowing down object default initialization.
If you don't mind, I prefer to keep an Init method (old way).

Reply:
happy with keeping an Init method?

Reply:
Sure